lib/services.js

```diff
@@ -90,11 +91,27 @@ module.exports = function generateServices(app, options) {
     { encoding: 'utf-8' }
   );
 
+  var commonModelPrefix = 'LoopBack';
+  if (options.namespaceCommonModels) {
+    commonModelPrefix = options.ngModuleName + options.namespaceDelimiter;
+    if (options.namespaceDelimiter === '.') {
+      return 'throw new Error("Unsupported delimiter "." for ' +
```

Comment:
Much better 👍 

As I was re-reading the code again, it struck me that it's throwing the error at the time the services.js file is loaded, instead of refusing to generate anything.

I personally prefer to report errors as soon as possible, i.e. crashing the generator when the configuration is not valid.

```diff
- return 'throw new Error("Unsupported delimiter "." for ' +
+ throw new Error('Unsupported delimiter "." for ' +
```

What are your arguments for deferring the error to the generated services.js file?

BTW I believe your code is not syntactically valid, see e.g. highlighting errors by GitHub below:

```js
throw new Error("Unsupported delimiter "." for ' +
  'namespacing common models.");
```

Please add a test to execute this branch to verify it works as intended.